### PR TITLE
Feat(eos_designs): Add support for no pasword in BGP sessions

### LINKED ISSUE #1198: eos_designs fails if BGP password are not set
### Issue Summary

In `eos_designs`, BGP must be configured with auth password unless ansible fails with the following error message:

```
fatal: [EAPI-LEAF1A -> localhost]: FAILED! =>
  msg: '''bgp_peer_groups'' is undefined'
```

But for some design, having BGP set with no auth could be useful. So `eos_designs` should support it

### Which component(s) of AVD impacted

eos_designs

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

```shell
Comment your BGP password in your inventory and run eos_designs playbook
```

### Relevant log output

```shell
fatal: [EAPI-LEAF1A -> localhost]: FAILED! =>
  msg: '''bgp_peer_groups'' is undefined'
```

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by titom73**
## Change Summary

Add condition to support no BGP password in `eos_designs/l3ls`

## Related Issue(s)

Fixes #1198

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Add `if/endif` statement to protect password knob in the following use-cases:

- underlay eBGP
- overlay eBGP
- overlay iBGP
- MLAG

## How to test
See Molecule `eos_designs-twodc-5stage-clos`: 

- DC1 with password
- DC2 with no password

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by ClausHolbechArista — COMMENTED** [review]
Looks good, except missing update of "fabric-variables.md" required->optional.
```yaml
# BGP peer groups encrypted password
# IPv4_UNDERLAY_PEERS and MLAG_IPv4_UNDERLAY_PEER | Required when < underlay_routing_protocol > == BGP
# EVPN_OVERLAY_PEERS | Required
```

**Review by ClausHolbechArista — APPROVED** [review]
LGTM!

**Comment by titom73** [comment]
> Looks good, except missing update of "fabric-variables.md" required->optional.
> 
> ```yaml
> # BGP peer groups encrypted password
> # IPv4_UNDERLAY_PEERS and MLAG_IPv4_UNDERLAY_PEER | Required when < underlay_routing_protocol > == BGP
> # EVPN_OVERLAY_PEERS | Required
> ```

Thanks. Documentation has been updated accordingly.

File: ansible_collections/arista/avd/molecule/eos_designs-twodc-5stage-clos/inventory/group_vars/DC1.yml [hunk 1]
```diff
@@ -1,6 +1,14 @@
 ---
 dc_name: DC1
 
+bgp_peer_groups:
+  IPv4_UNDERLAY_PEERS:
+    password: "AQQvKeimxJu+uGQ/yYvv9w=="
+  EVPN_OVERLAY_PEERS:
+      password: "q+VNViP5i4rVjW1cxFv2wA=="
```
Comment (emilarista) [inline ansible_collections/arista/avd/molecule/eos_designs-twodc-5stage-clos/inventory/group_vars/DC1.yml]: Nitpicking, but indentation looks off here.

Comment (titom73) [reply ansible_collections/arista/avd/molecule/eos_designs-twodc-5stage-clos/inventory/group_vars/DC1.yml]: Thanks ! Good catch. It is now fixed.

**Review by emilarista — APPROVED** [review]
LGTM!

**Review by onurgashi — APPROVED** [review]
LGTM

**Review by guillaumeVilar — COMMENTED** [review]
LGTM!

**Pull request merged**